Client.py

```diff
@@ -0,0 +1,227 @@
+import errno
+import os.path
+import socket
+
+import base
+import constants
+import CustomExceptions
+import FileObject
+import Request
+import util
+
+SUPPORTED_METHODS = ('GET')
+
+MIME_MAPPING = {
+    'html': 'text/html',
+    'png': 'image/png',
+    'txt': 'text/plain',
+}
+
+
+class Client(base.Base):
+    """ Client of Server, handle everything by itself
+        Variables:
+                self._socket = The socket of the client
+                self._recv_buff = any data we recieved but hadn't procssed yet
+                self._send_buff = All the data that needed to be sent but
+                                  hadn't sent yet
+                self._buff_size = The buff size of the server
+                self.file = FileObject of the file the request wants
+                self.request = Request object of the request we got
+    """
+
+    def __init__(
+        self,
+        s,
+        buff_size,
+        file_object=None,
+        request=None
+    ):
+        """
+        Client of Server, handle everything by itself
+        Arguements:
+        s: the socket of this client, CANNOT BE MODIFIED
+        _recv_buff: Every data we recv from the client and had not been
+                    procssed (mostly partial lines)
+        _send_buff: Every data we need to send but didn't sent yet
+        """
+        super(Client, self).__init__()
+        self._socket = s
+        self._recv_buff = ""
+        self._send_buff = ""
+        self._buff_size = buff_size
+        self.file = file_object
+        self.request = request
+
+    def get_socket(self):
+        """Get the socket of this client, private arguement"""
+        return self._socket
+
+    def _parse_header(self, line):
+        parsed = line.split(":", 1)
+        if len(parsed) == 2:
+            self.request.add_header(*parsed)
+            self.logger.debug("Added header, %s", line)
+
+    def recv(self):
```

Comment:
I can't be sure if I got the whole request at once. So this function making sure I receive what I need to get, when I need to get it.
Also recv process the data I get to the objects I need.